app/controllers/api/queries_controller.rb

```diff
@@ -62,11 +62,24 @@ def show
     def create
       options = {}
       options[:filters] = build_filter
-
+      
       authorize_providers
+      end_date = params[:effective_date]
+      start_date = params[:effective_start_date]
+
+      end_date = Time.at(end_date.to_i) if end_date.is_a?(String)
```

Comment:
There are some instances where the parameter passed in is a String, but some where it is a Time. That line is just ensuring that it's a Time for the following code, where I think .present? would run the code regardless and make it a bit redundant.